File: go-leetcode/skills/287.寻找重复数.go
package leetcode

func findDuplicate(nums []int) int {
	set := make(map[int]bool)

	for i := 0; i < len(nums); i++ {
		if !set[nums[i]] {
			set[nums[i]] = true
		} else {
			return nums[i]
		}
	}
	return 0
}

File: python-leetcode/skills/287.寻找重复数.py
from typing import List


class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        set_ = set()

        for i in range(len(nums)):
            if nums[i] not in set_:
                set_.add(nums[i])
            else:
                return nums[i]
        return 0
